dataRetriever/resolvers/validatorInfoResolver.go

```diff
@@ -124,6 +154,55 @@ func (res *validatorInfoResolver) resolveHashRequest(hash []byte, epoch uint32,
 	return res.marshalAndSend(data, pid)
 }
 
+// resolveMultipleHashesRequest sends the response for a hash array type request
+func (res *validatorInfoResolver) resolveMultipleHashesRequest(hashesBuff []byte, epoch uint32, pid core.PeerID) error {
+	b := batch.Batch{}
+	err := res.marshalizer.Unmarshal(&b, hashesBuff)
+	if err != nil {
+		return err
+	}
+	hashes := b.Data
+
+	validatorInfoForHashes, err := res.fetchValidatorInfoForHashes(hashes, epoch)
+	if err != nil {
+		return fmt.Errorf("resolveMultipleHashesRequest error %w from buff %s", err, hashesBuff)
```

Comment:
I don't think that the print for `hashesBuff` will work as expected, because it will display the string of each hash, but you would need the hex string of each hash